File: frontend/src/types/project.ts
import type { AuthSource } from './auth'

export interface Project {
	id: string
	name: string
	code: string
	is_active: boolean
	created_at: string
	updated_at: string
}

export interface ProjectMember {
	id: string
	username: string
	email: string | null
	status: string
	auth_source: AuthSource
	last_login_at: string | null
}

export interface ProjectCreateReq {
	name: string
	code: string
}

export interface ProjectUpdateReq {
	name: string
	code: string
}

export interface ProjectMemberReq {
	user_id: string
}

File: backend-go/internal/repository/model/project.go
package model

import "time"

type Project struct {
	Id        string    `db:"id"`
	Name      string    `db:"name"`
	Code      string    `db:"code"`
	IsActive  bool      `db:"is_active"`
	CreatedAt time.Time `db:"created_at"`
	UpdatedAt time.Time `db:"updated_at"`
}

type ProjectMember struct {
	ProjectId string    `db:"project_id"`
	UserId    string    `db:"user_id"`
	CreatedAt time.Time `db:"created_at"`
}
